File: index.server/Index.server/src/main/java/br/com/pedroxsqueiroz/Index/server/dtos/ContentChunckDto.java
package br.com.pedroxsqueiroz.Index.server.dtos;

import br.com.pedroxsqueiroz.Index.concept.models.Content;
import br.com.pedroxsqueiroz.Index.concept.models.ContentChunck;
import lombok.Data;

@Data
public class ContentChunckDto {

    private String id;
    private int start;
    private int end;
    private String chunckContent;
    private ContentDto content;

    public ContentChunckDto(ContentChunck chunck)
    {
        this.id     = chunck.getId();
        this.start  = chunck.getStart();
        this.end    = chunck.getEnd();
        this.content = new ContentDto(chunck.getContent());
    }

    public ContentChunckDto(ContentChunck chunck, ContentDto content)
    {
        this.id     = chunck.getId();
        this.start  = chunck.getStart();
        this.end    = chunck.getEnd();
        this.content = content;
    }

    public ContentChunckDto(ContentChunck chunck, String chunckContent, ContentDto content)
    {
        this(chunck, content);
        this.chunckContent = chunckContent;
    }

}

File: index.web_cli/src/dtos/content-chunck-dto.ts
import type ContentDto from "./content-dto";

export default class ContentChunckDto{
    
    constructor(
        id: string,
        start: number,
        end: number,
        chunckContent: string,
        content: ContentDto|null

    ){
        this.id = id;
        this.start = start;
        this.end = end;
        this.chunckContent = chunckContent;
        this.content = content;
    }

    id: string;
    start: number;
    end: number;
    chunckContent: string;    
    content: ContentDto|null;
}